core/tests/puppeteer-acceptance-tests/blog-admin-tests/delete-blog-post-draft.js

```diff
@@ -0,0 +1,21 @@
+const e2eBlogPostAdmin = require("../utility-functions/blogPostAdminUtils.js");
+const testConstants = require("../utility-functions/testConstants.js");
+
+const blogDashboardUrl = testConstants.URLs.BlogDashboard;
+const homePage = testConstants.Dashboard.LearnerDashboard;
+
+async function deleteDraftAsBlogPostAdmin() {
+  const blogPostAdmin = await new e2eBlogPostAdmin();
+  await blogPostAdmin.openBrowser();
+
+  await blogPostAdmin.signInWithEmail("testadmin@example.com");
+  await blogPostAdmin.waitForPageToLoad(homePage);
+  await blogPostAdmin.goto(blogDashboardUrl);
+  await blogPostAdmin.expectDraftBlogPostWithTitleToExist("Test Blog Post");
```

Comment:
Why would this exist when nothing has been done yet?

I am wondering if I'm confused about the setup steps. Is there already a blog post on the dashboard by this point? That might explain my confusion above.

If so, where do I find the order/sequence in which each of these tests gets run? Or are they supposed to be independent tests? (But if the latter is the case, then my question above applies, because I don't understand how the "test blog post" got created.)

Reply:
this test is specifically for testing the delete draft blog post, and I want to ask you if should we include the test dependencies in our tests? like in this one: first we need to test for creating a draft blog post(which I have done in some other file and created a draft blog with title 'Test Blog Post'), then we need to test this for deleting that draft in this file. By this we depends like to first create draft blog (so that the draft exists), and later we will test deleting draft by deleting that draft only!
or for not depending on other tests, we can test the create draft and delete draft in a single test file, this can be other way to do so!